core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -538,7 +538,10 @@ class PlaintextConsumerTest extends BaseConsumerTest {
   def testPositionAndCommit() {
     sendRecords(5)
 
-    assertNull(this.consumers.head.committed(new TopicPartition(topic, 15)))
+    // committed() on a partition that doesn't exist throws an exception
+    intercept[KafkaException] {
+      this.consumers.head.committed(new TopicPartition(topic, 15))
```

Comment:
I'm trying to think through the implications of this change. I'm a little concerned about inconsistency in the API. For example, if we receive an UNKNOWN_TOPIC_OR_PARTITION error from a fetch request, the consumer handles it silently (by requesting a metadata update). Whether or not we think that is ideal (it sucks IMHO), maybe it is worse to have different behavior depending on the API in question? So here's what I'm thinking and you can tell me if it makes sense or not.

1. When updating the internal offsets, we should catch UNKNOWN_TOPIC_OR_PARTITION, and if needed, set a flag in SubscriptionState noting that the partition doesn't exist and shouldn't be fetched. Later on, we can try to fetch offsets again. No error is surfaced, which is consistent with how we handle the same error from a fetch request.

2. The behavior of `committed()` and the commit APIs is tough to deal with. I think the existing code (without this patch) would return a null offset if the topic or partition doesn't exist (is that right?). However, for the commit APIs (`commitSync` and `commitAsync`), we currently raise a `KafkaException` to the user, which seems inconsistent. Maybe we use the behavior in this patch and let `committed()` also raise an exception if the topic doesn't exist?

So in total, the behavior would be to ignore non-existing partitions until the user tries to write to or access their committed state, at which point we raise an error. It's not ideal, but maybe it's a reasonable compromise given the way the consumer currently works.

An orthogonal issue I was thinking about is topic deletion. Consumer groups are not assigned non-existing partitions, but an existing partition could be deleted after assignment. We would typically expect to get the UNKNOWN_TOPIC_OR_PARTITION error code from a fetch request in that case. Currently, we just update metadata if that happens, but maybe we should instead try to rebalance the group? That would be consistent with the behavior if the topic doesn't exist at the time of assignment. (Note this is just for discussion, not something we should attempt here.)

cc @ijuma